### PR TITLE
[WIP] Fix transaction malleability by skipping over input signatures when computing the txid

### PR DESCRIPTION
**Pull request by bitcartel**
To fix transaction malleability, we want to make it no longer possible to take an existing valid transaction, modify the signature or sigscript of a transaction input, and re-submit the transaction to the network under a different txid - which can cause problems for building a chain on top of zero-conf transactions.

This patch introduces a new serialization flag SER_SKIP_TXIN_SIGSCRIPT.  When CTransaction is asked to generate its txid, this flag is used to instruct CTxIn to skip over it's sigscript field.

Tests
- zcash public alpha guide : OK
- zcash-gtest : OK
- test-bitcoin : nearly OK, the following tests fail (due to hardcoded data)
  - bloom_tests
  - miner_tests
  - script_tests
- qa/rpc-tests : TBD

### CODE REVIEW

**Comment by defuse** [comment]
Sorry, I renamed the wrong issue.

**Comment by ebfull** [comment]
We should also not hash `joinSplitSig`.

**Comment by ebfull** [comment]
Can we think of a way this change allows us to construct two transactions with the same `txid` which can simultaneously appear in the blockchain? My guess is it would be possible to construct a coinbase transaction of such a form... this surely breaks invariants.

File: src/serialize.h [hunk 1]
```diff
@@ -161,7 +161,7 @@ enum
     SER_NETWORK         = (1 << 0),
     SER_DISK            = (1 << 1),
     SER_GETHASH         = (1 << 2),
-    SER_SKIP_TXIN_SIGSCRIPT = (1 << 3),
+    SER_SKIP_SIGSCRIPT = (1 << 3),
```
Comment (daira) [inline src/serialize.h]: Minor nit: align ``= (1 << 3),`` with the lines above it.

**Comment by bitcartel** [comment]
Superceded by #1145.  Closing this PR.

**Pull request closed without merging**